packages/sanity/src/core/error/ErrorLogger.tsx

```diff
@@ -0,0 +1,67 @@
+import {useToast} from '@sanity/ui'
+import {useEffect} from 'react'
+import {ConfigResolutionError, SchemaError} from '../config'
+import {CorsOriginError} from '../store'
+import {globalScope} from '../util'
+
+const errorChannel = globalScope.__sanityErrorChannel
+
+/**
+ * Attaches a listener to the global error channel and displays a toast when a (unknown)
+ * error occurs. Compares the last displayed error message with the current
+ *
+ * @internal
+ */
+export function ErrorLogger(): null {
+  const {push: pushToast} = useToast()
+
+  useEffect(() => {
+    if (!errorChannel) return undefined
+
+    // errorChannel.subscribe() returns a unsubscriber function.
+    // By returning it from this `useEffect`, it'll unsubscribe on unmount.
+    return errorChannel.subscribe((msg) => {
+      // NOTE: Certain errors (such as the `ResizeObserver loop limit exceeded` error) is thrown
+      // by the browser, and does not include an `error` property. We ignore these errors.
+      if (!msg.error) {
+        return
+      }
+
+      // For errors that we "expect", eg have specific error screens for, do not push a toast
+      if (isKnownError(msg.error)) {
+        return
+      }
+
+      console.error(msg.error)
```

Comment:
Do we want to log the error even though it is a known error?

Reply:
I don't think so, since the error screens should have a much clearer explanation of what the error is.